openbas-model/src/main/java/io/openbas/database/repository/DocumentRepository.java

```diff
@@ -84,4 +84,18 @@ Optional<Document> findByIdGranted(
   @NotNull
   @EntityGraph(value = "Document.tags-scenarios-exercises", type = EntityGraph.EntityGraphType.LOAD)
   Page<Document> findAll(@NotNull Specification<Document> spec, @NotNull Pageable pageable);
+
+  @Query(
+      "SELECT DISTINCT d FROM Document d "
+          + "LEFT JOIN d.articles a "
+          + "LEFT JOIN d.injectDocuments i "
+          + "WHERE a.scenario.id = :scenarioId OR i.inject.scenario.id = :scenarioId")
```

Comment:
Why having condition on inject scenario id ? "ceinture et bretelles" ?

Reply:
It's because documents can be in injects (through the injectDocuments table) or articles. This was just easier than creating a findAllDistinctByArticleByScenarioId and findAllDistinctByInjectByScenarioId

Reply:
Add ink on document challenge

Reply:
Challenge documents added to the list.